gemini/use-cases/document-processing/doc_parsing_with_llamaindex_and_llamaparse.ipynb

```diff
@@ -0,0 +1,841 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6e52150147f8"
+      },
+      "source": [
+        "#### Load Nodes into Predefined Vector Store\n",
+        "\n",
+        "This following section required a preexisting Vertex AI Vector Store. Vector stores contain embedding vectors of ingested document chunks.\n",
+        "\n",
+        "For information to create a vector store, refer to this link https://docs.llamaindex.ai/en/stable/examples/vector_stores/VertexAIVectorSearchDemo/"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f8cdc3678a71"
+      },
+      "outputs": [],
+      "source": [
+        "vector_store = VertexAIVectorStore(\n",
+        "    project_id=PROJECT_ID,\n",
+        "    region=REGION,\n",
+        "    index_id=\"\",  # Add in your Vertex AI Vector Search Index ID\n",
+        "    endpoint_id=\"\",  # Add in your Vertex AI Vector Search Deployed Index ID\n",
+        "    gcs_bucket_name=GCS_BUCKET,\n",
+        ")\n",
+        "\n",
+        "# Only need to run once\n",
+        "vector_store.add(nodes)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "d22ef6b3a7e1"
+      },
+      "source": [
+        "#### Create a search index and search and query the Vector Store"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 26,
+      "metadata": {
+        "id": "c135cf0ee925"
+      },
+      "outputs": [],
+      "source": [
+        "lp_index = VectorStoreIndex.from_vector_store(vector_store)\n",
+        "lp_query_engine = lp_index.as_query_engine(similarity_top_k=2)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "7064bdd2c434"
+      },
+      "source": [
+        "## Query Comparison between LlamaIndex and LlamaParse\n",
+        "Below are queries that responses can be found in the 10Q document within complex tables. Let's see how each approach compares."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 27,
+      "metadata": {
+        "id": "bef0f00ab246"
+      },
+      "outputs": [],
+      "source": [
+        "queries = [\n",
+        "    \"What are the total cash, cash equivalents, and marketable securities as of Dec 23 2023\",\n",
+        "    \"Total investments with fair value change reflected in other comprehensive income as of Dec 23 2023\",\n",
+        "    \"What is the corporate debt securities unrealized loss as of Dec 31 2023 for 12 months or greater?\",\n",
+        "    \"What is the coupon rate for total outstanding debt\",\n",
+        "    \"Provide the table of share repurchases\",\n",
+        "]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 28,
+      "metadata": {
+        "id": "a0c410e45aea"
+      },
+      "outputs": [],
+      "source": [
+        "from termcolor import colored\n",
```

Comment:
Move this import to the top of the notebook

Reply:
Updated.